### PR TITLE
[WFLY-4198] - add null checks for MDC.contextMap and StoredLogDiagnostic...

### PR DESCRIPTION
**Pull request by baranowb**
...Context

MDC.getMap() might be null. Code assume it is never null.

https://issues.jboss.org/browse/WFLY-4198

### CODE REVIEW

File: ejb3/src/main/java/org/jboss/as/ejb3/component/interceptors/LogDiagnosticContextRecoveryInterceptor.java [hunk 1]
```diff
@@ -64,12 +64,14 @@ public static LogDiagnosticContextRecoveryInterceptor getInstance() {
     }
 
     public Object processInvocation(final InterceptorContext context) throws Exception {
-        for (String str : MDC.getMap().keySet()) {
-            MDC.remove(str);
+        if (MDC.getMap() != null) {
+            for (String str : MDC.getMap().keySet()) {
```
Comment (bstansberry) [inline ejb3/src/main/java/org/jboss/as/ejb3/component/interceptors/LogDiagnosticContextRecoveryInterceptor.java]: This is on the critical perf request path, and looking at the impl of MDC.getMap() it's pretty expensive, involving a map copy. So the result from the getMap() call at L67 should be cached and reused in L68.

**Comment by baranowb** [comment]
@bstansberry  done + rebase.

File: ejb3/src/main/java/org/jboss/as/ejb3/component/interceptors/LogDiagnosticContextStorageInterceptor.java [hunk 2]
```diff
@@ -62,11 +62,15 @@ public static LogDiagnosticContextStorageInterceptor getInstance() {
     }
 
     public Object processInvocation(final InterceptorContext context) throws Exception {
-        context.putPrivateData(KEY, new StoredLogDiagnosticContext(MDC.getMap(), NDC.get()));
-        try {
+        if(MDC.getMap() != null){
```
Comment (bstansberry) [inline ejb3/src/main/java/org/jboss/as/ejb3/component/interceptors/LogDiagnosticContextStorageInterceptor.java]: Same thing applies here.

**Comment by bstansberry** [comment]
Superceded by #7754

**Pull request closed without merging**